pkg/fdbadminclient/mock/admin_client_mock.go

```diff
@@ -28,58 +28,59 @@ import (
 	"sync"
 	"time"
 
+	"github.com/FoundationDB/fdb-kubernetes-operator/pkg/fdbadminclient"
+
 	fdbv1beta2 "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta2"
 	"github.com/FoundationDB/fdb-kubernetes-operator/internal"
-	"github.com/FoundationDB/fdb-kubernetes-operator/pkg/fdbadminclient"
 	"github.com/FoundationDB/fdb-kubernetes-operator/pkg/podmanager"
 	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// mockAdminClient provides a mock implementation of the cluster admin interface
-type mockAdminClient struct {
+// AdminClient provides a mock implementation of the cluster admin interface
+type AdminClient struct {
 	Cluster                                  *fdbv1beta2.FoundationDBCluster
 	KubeClient                               client.Client
 	DatabaseConfiguration                    *fdbv1beta2.DatabaseConfiguration
 	ExcludedAddresses                        []string
 	ReincludedAddresses                      map[string]bool
 	KilledAddresses                          []string
-	frozenStatus                             *fdbv1beta2.FoundationDBStatus
+	FrozenStatus                             *fdbv1beta2.FoundationDBStatus
 	Backups                                  map[string]fdbv1beta2.FoundationDBBackupStatusBackupDetails
 	restoreURL                               string
 	clientVersions                           map[string][]string
 	missingProcessGroups                     map[string]bool
 	additionalProcesses                      []fdbv1beta2.ProcessGroupStatus
 	localityInfo                             map[string]map[string]string
 	incorrectCommandLines                    map[string]bool
-	maxZoneFailuresWithoutLosingData         *int
-	maxZoneFailuresWithoutLosingAvailability *int
-	knobs                                    []string
-	maintenanceZone                          string
+	MaxZoneFailuresWithoutLosingData         *int
+	MaxZoneFailuresWithoutLosingAvailability *int
```

Comment:
What is the convention here? When to do the capitalization and when not to?

Reply:
In go the capitalization is used to make a variable/method publicly exposed. All variables that are accesses from the outside have a capital letter.